cryptarithm.py

```diff
@@ -13,135 +13,111 @@
+            self.lower_bound = 1
+            self.upper_bound = 9
         else:
-            self.domain = tuple(range(10))
+            self.var = Integer(self.label, lower_bound=0, upper_bound=9)
+            self.lower_bound = 0
+            self.upper_bound = 9
 
-    def __repr__(self):
-        return "(name: {name}, coefficient: {coeff}, domain: {domain})".format(name=self.name,
-                                                                               coeff=self.coefficient,
-                                                                               domain=self.domain)
 
-
-def build_dqm(variable_list: List[LetterVariable], coefficient_map: dict) -> DQM:
-    """Build the discrete quadratic model from provided variable list.
+def build_cqm(model_variables: List[ModelVariable]) -> CQM:
+    """Build a CQM model for the verbal arithmetic problem.
 
     Args:
-        variable_list: List of variables for cryptarithm problem.
-        coefficient_map: Dictionary mapping variables to their coefficients.
-    
-    Returns:
-        dqm: Corresponding discrete quadratic model. 
+        model_variables: A list of model variables that define the problem.
 
+    Returns:
+        A CQM model for the verbal arithmetic problem.
+    
     """
-    dqm = DQM()
-
-    # Scaling value arbitrarily chosen to tame energies for larger problems.
-    eq_constr_scale = 1/(2**(len(coefficient_map)))
-
-    # Set linear biases from equality constraint
-    print("setting linear biases...")
-    for variable in variable_list:
-        dqm.add_variable(len(variable.domain), variable.name)
-        for idx in range(len(variable.domain)):
-            dqm.set_linear_case(variable.name, idx, eq_constr_scale*
-                                                    (coefficient_map[variable.name]*
-                                                    variable.domain[idx])**2)
-
-    # Set quadratic biases from equality constraint
-    print("setting quadratic biases...")
-    for var1, var2 in combinations(variable_list, r=2):
-        for i in range(dqm.num_cases(var1.name)):
-            for j in range(dqm.num_cases(var2.name)):
-                dqm.set_quadratic_case(
-                    var1.name, i, var2.name, j, 2*
-                                                eq_constr_scale*
-                                                var1.domain[i]*
-                                                var2.domain[j]*
-                                                coefficient_map[var1.name]*
-                                                coefficient_map[var2.name]
-                )
-
-    # Choose a large penalty for quadratic interactions in same states
-    quad_penalty = eq_constr_scale*max(np.absolute(dqm.to_numpy_vectors()[2][2]))
+    cqm = CQM()
+
+    # Both sides equal constraint
+    cqm.add_constraint(
+        quicksum(
+            [variable.coefficient*variable.var for variable in model_variables]
```

Comment:
better to use a generator than a list
```suggestion
            (variable.coefficient*variable.var for variable in model_variables)
```

for performance, it's even better to use labels directly rather than `quicksum`, but I like `quicksum` here for pedagogy.